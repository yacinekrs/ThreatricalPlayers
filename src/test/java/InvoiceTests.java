import org.junit.jupiter.api.Test;
import java.util.List;
import java.util.Map;
import static org.approvaltests.Approvals.verify;

public class InvoiceTests {

    @Test
    void toTextTest() {
        Map<String, Play> playsMap = Map.of(
                "hamlet", new Tragedie("Hamlet"),
                "as-like", new Comedie("As You Like It"),
                "othello", new Tragedie("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1123AB", 150), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)),playsMap);

        String result = invoice.toText();
        verify(result);
    }
    
    @Test
    void toTextTestComedieInf() {
        Map<String, Play> playsMap = Map.of(
                "hamlet", new Tragedie("Hamlet"),
                "as-like", new Comedie("As You Like It"),
                "othello", new Tragedie("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1123AB", 150), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 15),
                new Performance("othello", 40)),playsMap);

        String result = invoice.toText();
        verify(result);
    }

    @Test
    void toTextTestTragedieInf() {
        Map<String, Play> playsMap = Map.of(
                "hamlet", new Tragedie("Hamlet"),
                "as-like", new Comedie("As You Like It"),
                "othello", new Tragedie("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1123AB", 150), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 25)),playsMap);

        String result = invoice.toText();
        verify(result);
    }

    @Test
    void toTextTestSoldPntInf() {
        Map<String, Play> playsMap = Map.of(
                "hamlet", new Tragedie("Hamlet"),
                "as-like", new Comedie("As You Like It"),
                "othello", new Tragedie("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1123AB", 10), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 25)),playsMap);

        String result = invoice.toText();
        verify(result);
    }
    
    @Test
    void toHTMLTest() {
        Map<String, Play> playsMap = Map.of(
                "hamlet", new Tragedie("Hamlet"),
                "as-like", new Comedie("As You Like It"),
                "othello", new Tragedie("Othello"));
       
        Invoice invoice = new Invoice(new Customer("BigCo", "1123AB", 150), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)),playsMap);

        String result = invoice.toHTML();
        verify(result);
    }
}
