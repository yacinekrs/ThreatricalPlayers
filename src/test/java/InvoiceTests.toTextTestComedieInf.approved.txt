Statement for BigCo
  Hamlet: $650.00 (55 seats)
  As You Like It: $345.00 (15 seats)
  Othello: $500.00 (40 seats)
Amount owed is $1,480.00
You earned 38 credits